### ISSUE TITLE
`unfulfilled_lint_expectations` incorrectly triggered by `expect(clippy::used_underscore_items)`

### ISSUE THREAD

**Issue by meditationmind** [rust-lang/rust-clippy#17650]
### Summary

I suspect this is caused by [#17308](https://github.com/rust-lang/rust-clippy/pull/17308).

Prior to `nightly-2026-08-25`, `clippy::used_underscore_items` was not triggered by this use case (`Self::_f()`), so the `expect` was not needed.

### Lint Name

unfulfilled_lint_expectations / used_underscore_items

### Reproducer

I tried this code:

```rust
#![warn(clippy::used_underscore_items)]

struct X;

impl X {
    fn f() {
        Self::_f();
    }

    #[expect(clippy::used_underscore_items)]
    fn _f() {}
}

fn main() {
    X::f();
}
```

I expected to see this happen:

Successfully compile with no errors or warnings.

Instead, this happened:

`unfulfilled_lint_expectations` is triggered, but if the `#[expect()]` is removed, the code fails to compile:

```rust
error: used underscore-prefixed item                                                                                                                                                                                     
  --> src\main.rs:7:9
   |
 7 |         Self::_f();
   |         ^^^^^^^^
   |
note: item is defined here
  --> src\main.rs:11:5
   |
11 |     fn _f() {}
   |     ^^^^^^^
   = help: for further information visit https://rust-lang.github.io/rust-clippy/main/index.html#used_underscore_items
   = note: `-D clippy::used-underscore-items` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::used_underscore_items)]`
```

### Version

```text
rustc 1.100.0-nightly (17fd5b8a3 2026-08-28)
binary: rustc
commit-hash: 17fd5b8a37b6667b6cc137f3cc35f09759768a3b
commit-date: 2026-08-28
host: x86_64-pc-windows-msvc
release: 1.100.0-nightly
LLVM version: 23.1.0
```

**Comment by Jarcho** [comment]
I can't reproduce this on the playground (2026-08-26).

**Comment by meditationmind** [comment]
That's so weird, I can't reproduce now either. Maybe it was some kind of cargo caching issue since I use the stable toolchain for most of my projects. Sorry for the false alarm! I will close this.
